ksqldb-engine/src/main/java/io/confluent/ksql/query/SerializationClassifier.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.query;
+
+import io.confluent.ksql.query.QueryError.Type;
+import io.confluent.ksql.util.ReservedInternalTopics;
+import java.util.Objects;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.streams.errors.StreamsException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@code SerializationClassifier} classifies serialization exceptions as user error
+ * while writing to a user topic due to schema mismatch
+ */
+public class SerializationClassifier implements QueryErrorClassifier {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SerializationClassifier.class);
+
+  private final String queryId;
+
+  public SerializationClassifier(final String queryId) {
+    this.queryId = Objects.requireNonNull(queryId, "queryId");
+  }
+
+  private boolean hasInternalTopicPrefix(final Throwable e) {
+    final int index = ExceptionUtils.indexOfThrowable(e, SerializationException.class);
+    final String errMsg = ExceptionUtils.getThrowableList(e).get(index).getMessage();
+
+    return errMsg.contains("Error serializing message to topic: "
```

Comment:
Maybe startsWith if we are certain its in the beginning of the message?

Reply:
Looking at the topic name and performing a startsWith instead of the message.